pkg/services/object/get/exec.go

```diff
@@ -118,7 +119,16 @@ func (exec execCtx) key() (*ecdsa.PrivateKey, error) {
 		}
 	}
 
-	return exec.svc.keyStore.GetKey(sessionInfo)
+	key, err := exec.svc.keyStore.GetKey(sessionInfo)
+	if err != nil && errors.As(err, new(apistatus.SessionTokenNotFound)) {
+		// a new request with session token should be
+		// spawned but no session token has been found
+		// so try to use node's key as the best effort
+		exec.prm.common.ForgetTokens()
+		key, err = exec.svc.keyStore.GetKey(nil)
```

Comment:
I don't see this logic being limited to some specific types of requests. And it makes the node sign the request with its own key in some cases. Can this be misused (specifically triggering the node to sign some requests)?

Reply:
Can we make it for range-hash only? Just move this check in `GetRangeHash`.

Yes, it looks ugly, but the problem is deep -- we have 4 types of requests being processed by the same code. And in each function we are switching and doing different things based on the request type (actually, returning `Object` for `GetRange` and making `GetRange` instead of `GetRangeHash` are two most obvious examples). I am doing some simplification refactoring currently, will create a PR after release.

Reply:
To clarify, there is no problem for other requests because they can setup request forwarder.

Reply:
They probably can, but I'm more concerned about situations when they deliberately do not do that and then have (or not?) the node using its key to sign something. Can this be leveraged for some privilege escalation when the request originator doesn't have appropriate permission while the node does?

Reply:
@roman-khimov, @fyrchik, yes, the fix isn't perfect and yes, the problem is deeper than just allowing static sessions for range requests, requires a refactor and/or an arch revision.

> I don't see this logic being limited to some specific types of requests.

> Can we make it for range-hash only?

@fyrchik, @roman-khimov yes, we can but what for? if `func (exec execCtx) key()` can't get the session token, attaching tokens is useless for _any_ request _the node_ signs (ok, except the case when a requester attached a _static_ session _with a node_, could not imagine when that is possible). We either fail here or try to do our best (the test from the issue asks us to fix it now).

> Can this be misused (specifically triggering the node to sign some requests)?

@roman-khimov, well, it could be used somehow i guess but can not imagine any practical case: the changes are for "get" requests only, and new requests are spawned only when a node tries to fetch an object (that means that a node has already agreed about a request (the original one, with the original tokens and the original requester) and allows it in terms of our (e)ACL rules).

Reply:
`SessionNotFound` error is much more descriptive than `ObjectNotFound` or `ObjectAccessDenied`. Also, we do not spawn unnecessary network requests in this case.

Reply:
ok, moved